### DISCUSSION TITLE
ansible install milvus cluster

### DISCUSSION THREAD

**Discussion by Anandonzy** [milvus-io/milvus#17957] [Q&A and General discussion]
https://milvus.io/cn/docs/v2.0.x/install_cluster-ansible.md#Configure-deploy-dockeryml
When I see this document. I  try to install milvus cluster by ansible.
`ansible-playbook /home/app/milvus-master/deployments/docker/cluster-distributed-deployment/deploy-docker.yml`
Then it is not ok.
`PLAY [setup pre-requisites] *****************************************************************************************************

TASK [Gathering Facts] **********************************************************************************************************
ok: [10.24.18.71]
ok: [10.24.18.60]
ok: [10.24.18.59]

TASK [install-modules : Install Packages] ***************************************************************************************
[DEPRECATION WARNING]: Invoking "apt" only once while using a loop via squash_actions is deprecated. Instead of using a loop to 
supply multiple items and specifying `name: "{{ item }}"`, please use `name: ['ntp', 'python3', 'tcpdump', 'wget', 'openssl', 
'curl', 'python3-pip']` and remove the loop. This feature will be removed in version 2.11. Deprecation warnings can be disabled 
by setting deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: Invoking "apt" only once while using a loop via squash_actions is deprecated. Instead of using a loop to 
supply multiple items and specifying `name: "{{ item }}"`, please use `name: ['ntp', 'python3', 'tcpdump', 'wget', 'openssl', 
'curl', 'python3-pip']` and remove the loop. This feature will be removed in version 2.11. Deprecation warnings can be disabled 
by setting deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: Invoking "apt" only once while using a loop via squash_actions is deprecated. Instead of using a loop to 
supply multiple items and specifying `name: "{{ item }}"`, please use `name: ['ntp', 'python3', 'tcpdump', 'wget', 'openssl', 
'curl', 'python3-pip']` and remove the loop. This feature will be removed in version 2.11. Deprecation warnings can be disabled 
by setting deprecation_warnings=False in ansible.cfg.
[WARNING]: Updating cache and auto-installing missing dependency: python-apt
failed: [10.24.18.71] (item=[u'ntp', u'python3', u'tcpdump', u'wget', u'openssl', u'curl', u'python3-pip']) => {"ansible_loop_var": "item", "changed": false, "cmd": "apt-get update", "item": ["ntp", "python3", "tcpdump", "wget", "openssl", "curl", "python3-pip"], "msg": "[Errno 2] No such file or directory", "rc": 2}
failed: [10.24.18.60] (item=[u'ntp', u'python3', u'tcpdump', u'wget', u'openssl', u'curl', u'python3-pip']) => {"ansible_loop_var": "item", "changed": false, "cmd": "apt-get update", "item": ["ntp", "python3", "tcpdump", "wget", "openssl", "curl", "python3-pip"], "msg": "[Errno 2] No such file or directory", "rc": 2}
failed: [10.24.18.59] (item=[u'ntp', u'python3', u'tcpdump', u'wget', u'openssl', u'curl', u'python3-pip']) => {"ansible_loop_var": "item", "changed": false, "cmd": "apt-get update", "item": ["ntp", "python3", "tcpdump", "wget", "openssl", "curl", "python3-pip"], "msg": "[Errno 2] No such file or directory", "rc": 2}

PLAY RECAP **********************************************************************************************************************
10.24.18.59                : ok=1    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
10.24.18.60                : ok=1    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
10.24.18.71                : ok=1    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0 `

What I should do. Thanks!

**Comment by Anandonzy** [comment]
hello

**Reply by john-h-luo** [reply]
Hi, this issue is caused by the version diference of ansible, I include a quick fix and a reference URL here.
Quick fix: modify the code in yaml file to something like listed below, file path: /milvus/deployments/docker/cluster-distributed-deployment/roles/install-modules/tasks/main.yml

---
- name: Install Packages 
  apt: 
    pkg:
      - ntp
      - python3
      - tcpdump
      - wget
      - openssl
      - curl
      - python3-pip
    state: present

Reference: https://github.com/geerlingguy/ansible-role-docker/issues/77

**Reply by Anandonzy** [reply]
Thank you,Now I should install your list Packages.I have a try.

**Reply by Anandonzy** [reply]
This file  /milvus/deployments/docker/cluster-distributed-deployment/roles/install-modules/tasks/main.yml is tell me what packages I need to install?which version ? 
Do you have a detailed installation instructions.

**Reply by john-h-luo** [reply]
Would you please show me the the original message?

**Reply by Anandonzy** [reply]
[app@l18-71-p-pbs milvus-master]$ pip install ntp
Collecting ntp
  Downloading https://files.pythonhosted.org/packages/b6/8e/5afd387fda86864c486c369e12df15e12e1f22bae70004a16f235bb590ba/ntp-1.0.0a1.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-anC7Co/ntp/setup.py", line 11, in <module>
        version_str = open(os.path.join('ntp', '_version.txt'), 'r').read().strip()
    IOError: [Errno 2] No such file or directory: 'ntp/_version.txt'
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-anC7Co/ntp/
You are using pip version 8.1.2, however version 22.1.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.

**Reply by Anandonzy** [reply]
<img width="1016" alt="image" src="https://user-images.githubusercontent.com/32696359/176659605-226183ef-a359-4d19-9bec-270e081b0aa4.png">

**Reply by john-h-luo** [reply]
It seems like you have a out-dated version of pip installed on this machine, you must either update or remove it before newer version can be installed which Ansible depends on.

**Reply by Anandonzy** [reply]
Now I upgrade my pip. This pakeage  need to use pip3 install?

**Reply by john-h-luo** [reply]
I can't answer this question without knowing what step you are at, please join us on the Slack channel where you can ask question and get response much faster. https://milvus.io/community
